src/fidesops/service/connectors/saas_query_config.py

```diff
@@ -144,19 +180,62 @@ def map_param_values(
                 f"Unable to replace placeholders in body for the '{self.action}' request of the '{self.collection_name}' collection."
             )
 
+        # format the body based on the content type
+        updated_headers, formatted_body = self.format_body(headers, body)
+
         return SaaSRequestParams(
             method=current_request.method,
             path=path,
-            headers=headers,
+            headers=updated_headers,
             query_params=query_params,
-            json_body=json.loads(body) if body else update_values,
+            body=formatted_body,
         )
 
+    @staticmethod
+    def format_body(
```

Comment:
This is my proposal for keeping the body formatting in one place. I know we've had issues with payloads being the wrong type (string vs JSON).

Reply:
makes sense to me -